test/unit/carrier_test.rb

```diff
@@ -84,6 +84,24 @@ def test_timestamp_from_business_day_returns_fifteen_days_in_the_future
     end
   end
 
+  def test_timestamp_from_business_day_handles_saturday
+    current = DateTime.new(2016, 7, 9) # Saturday
+    expected = DateTime.new(2016, 7, 11)
+
+    Timecop.freeze(current) do
+      assert_equal expected, @carrier.send(:timestamp_from_business_day, 1)
+    end
+  end
+
+  def test_timestamp_from_business_day_handles_saturday
```

Comment:
It's actually an active support feature, but this gem depends on it explicitly so that should be fine.